src/System.Net.Http/src/System/Net/Http/HttpContent.cs

```diff
@@ -142,7 +142,9 @@ internal void SetBuffer(byte[] buffer, int offset, int count)
 
         internal bool TryGetBuffer(out ArraySegment<byte> buffer)
         {
-            return _bufferedContent != null && _bufferedContent.TryGetBuffer(out buffer);
+            bool result = _bufferedContent != null && _bufferedContent.TryGetBuffer(out buffer);
+            Debug.Assert(result, "Performance: TryGetBuffer failed, which means that reads from the content will copy the data!");
```

Comment:
@davidsh How would we go about logging this message, then? I looked around the code a bit and found [NetEventSource.PrintError](https://github.com/jamesqo/corefx/blob/784a167a96ce69e245eed808c0e42c3401b13101/src/Common/src/System/Net/Logging/NetEventSource.cs#L216), but it calls a function named `CriticalError` so I'm not really sure if it's appropriate for this.